File: javascript/decode.js
function decode(tree,str){
	var output = ''
	var p = tree
	for(bit in str){
		if(str[bit] == '0'){p = p[0];}
		else{p = p[1];}
		if(typeof(p) == 'string'){
			output += p
			p = tree
		}
	}
	return output;
}
console.log(decode([['a', ['g', 'c']], [['f', ['b', 'd']], 'e']],'000000101001101110111111111111100100010'));

File: python/decode.py
def decode(tree,str):
	output = ''
	p = tree
	for bit in str:
		if bit == '0': p = p[0]
		else: p = p[1]
		if type(p) == type('') :
			output += p
			p = tree
	return output
print decode((('a', ('g', 'c')), (('f', ('b', 'd')), 'e')),'000000101001101110111111111111100100010')
